src/hmm/topology.h

```diff
@@ -0,0 +1,138 @@
+// hmm/topology.h
+
+// Copyright 2009-2011  Microsoft Corporation
+//                2019  Johns Hopkins University (author: Daniel Povey)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef KALDI_HMM_HMM_TOPOLOGY_H_
+#define KALDI_HMM_HMM_TOPOLOGY_H_
+
+#include <fst/fstlib.h>
+#include "base/kaldi-common.h"
+
+
+namespace kaldi {
+
+
+/// \addtogroup hmm_group
+/// @{
+
+/*
+  The following would be the text form for the "normal" 3-state HMM topology/
+  "bakis model", with the typical reordering that we do to improve the
+  compactness of the compiled FSTs.  The format is the OpenFst acceptor format.
+  The fields are, for transitions,
+  <from-state> <to-state> <pdf-class> <transition-cost>
+ and, for final-states,
+  <state> <final-cost>
```

Comment:
@danpovey @hhadian I seriously think that we should restrain each FST in the Topology class to have only a single final state. Otherwise, it is not clear how we would define MinLength() in a valid way.